dino: 2019-10-28 -> 0.1.0

alyssais opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

alyssais committed:
dino: 2019-10-28 -> 0.1.0

The system libsignal-protocol-c is preferred now, so fetching
submodules is no longer required.

The other libraries added are ones that the build complained about.  I
suspect they are optional dependencies.  Adding them makes the build
quieter, and has basically no impact on runtime closure size, so no
harm in making them available to keep the build system happy.

alyssais committed:
dino: drop unnecessary enableParallelBuilding

This is the default.

alyssais committed:
dino: add myself as a maintainer

andir reviewed (approved):
Works like a charm for me. I really like the new read & delivery markers :)

alyssais commented:
@GrahamcOfBorg eval

alyssais committed:
dino: fix rendering of logo in about window

Previously:

** (dino:7442): CRITICAL **: 00:35:22.373: file /build/source/main/src/ui/unified_window.vala: line 86: uncaught error: Unrecognized image file format (gdk-pixbuf-error-quark, 3)

(dino:7442): Gtk-WARNING **: 00:35:22.601: Found an icon but could not load it. Most likely gdk-pixbuf does not provide SVG support.

andir commented:
Interestingly I didn't run into that about dialog issue. It just shows here. What platform are you running this on?

orivej commented on pkgs/applications/networking/instant-messengers/dino/default.nix:
```diff
@@ -51,12 +52,19 @@ stdenv.mkDerivation {
+    libsepol
+    libpsl
+    brotli
+    libsignal-protocol-c
```
These (except `libsignal-protocol-c`) are private dependencies of the actual dependencies that should neither be propagated nor added manually to dino. For example, libpsl and brotli come from libsoup-2.4.pc:
```
Requires: glib-2.0 >=  2.38, gobject-2.0 >=  2.38, gio-2.0 >=  2.38
Requires.private: libxml-2.0, sqlite3, libpsl >=  0.20, libbrotlidec, zlib
```
(To be precise, `glib` uses `utillinuxMinimal` rather than `utillinux`.)

The warnings we see, such as `Package 'mount', required by 'gio-2.0', not found`, come from CMake running both `pkg-config --libs` and `pkg-config --static --libs` to populate both `<XXX>_LIBRARIES` and `<XXX>_STATIC_LIBRARIES` ([docs](https://cmake.org/cmake/help/v3.16/module/FindPkgConfig.html)), but dino has no use for the latter. Currently these warnings can not be disabled: https://gitlab.kitware.com/cmake/cmake/issues/18158

(They could be prevented by pruning `Requires.private` from shared-only libraries akin to #51767, although it can not be detected if a library is shared-only from the .pc file alone, and this is just a warning.)

orivej commented:
> Interestingly I didn't run into that about dialog issue. It just shows here.

Me too. It should not be necessary to list `librsvg` explicitly because even without it the resulting dino binary wrapper contains GDK_PIXBUF_MODULE_FILE=…librsvg….

alyssais commented:
> Interestingly I didn't run into that about dialog issue. It just shows
> here. What platform are you running this on?

NixOS.

alyssais commented:
> These (except `libsignal-protocol-c`) are private dependencies of the
> actual dependencies that should neither be propagated nor added
> manually to dino.

Looks like this was merged a bit prematurely, since this hadn't been
addressed.  I'll open a PR to remove them again.
